### DISCUSSION TITLE
Improve choice of cheapest default carrier in front office

### DISCUSSION THREAD

**Discussion by Hlavtox** [PrestaShop/PrestaShop#37664] [Ideas]
### The problem

Most of my clients have PrestaShop configured to choose the cheapest delivery option as the default one.

But, there is a catch. ALL of stores I manage have an option to personally pickup the goods at the store or warehouse, which is always free.

So, the default delivery option the customer gets shown after adding something to cart is personal pickup = shows Free shipping in the cart modal and on cart page. Only for the customer to discover that he actually needs to pay shipping later.

### The solution

I resolve this for my clients using hook `actionFilterDeliveryOptionList`, where I re-calculate the `is_best_price` without personal pickup.

But, I think we should consider adding this to the core.

Idea 1 - Add checkbox to carrier, something like "Exclude this carrier when calculating best price".
Idea 2 - Add a "Carrier type" radio with options "Classic/Personal pickup" to carrier? Maybe Prestashop could then calculate the cheapest carrier only from the Classic ones. Or we could add a new option to `Shipping > Configuration > Default carrier` called `Cheapest classic carrier`.
Idea 3 - Add new option to `Shipping > Configuration > Default carrier` called `Cheapest from following`, and you select the ones to be the default pool

### Additional context

cc @kpodemski @ibahloul-ps 

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by MateoCastilloPS** [comment]
Hi @Hlavtox 👋 

Great pain pont you mention here, and as you stated in [this discussion](https://github.com/PrestaShop/PrestaShop/discussions/37665), I think the best option is to set carrier types, that not only would help solve this issue, but, I think also would help to improve some other parts of the software in the future.

What do you think if we go a bit further and we split what you called "Classic" to have in total set 3 types of delivery : 

1. Personal pickup 
2. Delivery to an address
3. Pickup and Drop off Point

The idea having the #3 option is to be able to identify the carriers that need to display a map (via a module), this could be useful in the future to help improving how the delivery addresses for pickup points are gathered and stocked in PrestaShop (I know there are some pains about it too) 

(interested in your opinion also @Matt75 @prestascott ^^)

**Reply by Hlavtox** [reply]
Hey! Cool! :-)

For this issue, only two groups are needed, but maybe you will find some other issues that more "groups" could resolve.

For the map selection - this is handled by `displayCarrierExtraContent`, it works for well for a long time, wouldn't touch it. 😄
